src/app/src/test/kotlin/org/imperial/mrc/hint/integration/ADRTests.kt

```diff
@@ -56,6 +56,31 @@ class ADRTests : SecureIntegrationTests() {
         }
     }
 
+    @ParameterizedTest
+    @EnumSource(IsAuthorized::class)
+    fun `can get ADR datasets`(isAuthorized: IsAuthorized) {
+        testRestTemplate.postForEntity<String>("/adr/key", getPostEntity())
+        val result = testRestTemplate.getForEntity<String>("/adr/datasets")
+
+        assertSecureWithSuccess(isAuthorized, result, null)
+
+        if (isAuthorized == IsAuthorized.TRUE) {
+            val data = ObjectMapper().readTree(result.body!!)["data"]
+            assertThat(data.isArray).isTrue()
+            // we used a fake key so no results will come back
```

Comment:
Does this mean that if a user enters their key wrongly they won't see any sort of error, but will just see no results returned? Maybe we should handle that in the front end (or with a displayed error message originating in the backend) - "No datasets found - this can be caused by an incorrect ADR key."  One for another ticket?

Reply:
In the mockups we've now added a validation step when the key is first added. Not implemented yet but it should prevent saving if the key isn't valid (by hitting some ADR endpoint to check validity.)